### ISSUE TITLE
[BUG] Benchmark / Fetch - process still hanging

### ISSUE THREAD

**Issue by zekth** [denoland/deno_std#316]
see: https://github.com/denoland/deno_std/pull/312

EDIT:
After few checks it seems like using `https://google.com/` instead of `https://deno.land/` works but using `https://www.google.com/` fails.

So it seems like fetch is not handling well the redirections.

cc @athecoder

**Comment by ATheCoder** [comment]
This makes deno v0.3.5 hang after printing `Finished`
```
(async () => {
  const urls = new Array(10).fill("https://deno.land/");
    await Promise.all(urls.map((denoland: string) => fetch(denoland)));
    console.log("Finished!");
})();
```
Changing `Array.fill(10)` to `Array.fill(9)` makes deno hang sometimes and exit other times.
Changing `Array.fill(10)` to Array.fill(8)` makes deno exit everytime(?)

**Comment by ATheCoder** [comment]
Problem persists with deno v0.3.4, but is completely fixed on v0.3.3

**Comment by zekth** [comment]
Fix merged. Closing
